packages/material-ui/src/internal/SwitchBase.d.ts

```diff
@@ -21,6 +21,11 @@ export interface SwitchBaseProps
   value?: unknown;
 }
 
+export interface SwitchBaseComponentProps extends React.HTMLAttributes<HTMLInputElement> {
+  // Accommodate arbitrary additional props coming from the `inputProps` prop
+  [arbitrary: string]: any;
```

Comment:
The problem is that this will not catch excess props that are actually wrong (such as those from typos or accidental wrong names.